python/ray/_common/utils.py

```diff
@@ -214,27 +220,112 @@ def get_call_location(back: int = 1):
+            node_infos = gcs_client.get_all_node_info(
+                filters=[
+                    ("node_id", "=", node_id),
+                    ("state", "=", "ALIVE"),
+                ]
+            ).values()
+        except Exception as e:
+            raise Exception(
+                f"Failed to get node info from GCS when fetching tempdir for node {node_id}: {e}"
+            )
+
+        # Check that node infos is not empty
+        if not node_infos:
+            raise Exception(
+                f"No node info associated with ALIVE state found for node {node_id} in GCS"
+            )
+
+        # Get the first node info
+        node_info = next(iter(node_infos))
+
+        if node_info is not None:
+            temp_dir = getattr(node_info, "temp_dir", None)
+            if temp_dir is not None:
+                return temp_dir
+            else:
+                logger.warning(
+                    "Node temp_dir not found in NodeInfo. "
+                    "Using Ray's default temp dir."
+                )
+
+    # fallback to default ray temp dir
+    tmp_dir = None
     if "RAY_TMPDIR" in os.environ:
-        return os.environ["RAY_TMPDIR"]
+        tmp_dir = os.environ["RAY_TMPDIR"]
     elif sys.platform.startswith("linux") and "TMPDIR" in os.environ:
-        return os.environ["TMPDIR"]
+        tmp_dir = os.environ["TMPDIR"]
     elif sys.platform.startswith("darwin") or sys.platform.startswith("linux"):
         # Ideally we wouldn't need this fallback, but keep it for now for
         # for compatibility
-        tempdir = os.path.join(os.sep, "tmp")
+        tmp_dir = os.path.join(os.sep, "tmp")
     else:
-        tempdir = tempfile.gettempdir()
-    return tempdir
+        tmp_dir = tempfile.gettempdir()
+
+    return os.path.join(tmp_dir, "ray")
+
+
+def get_user_temp_dir():
+    """
+    get_user_temp_dir is deprecated. Use resolve_user_ray_temp_dir instead.
+
+    Get the ray temp directory for the current user.
+
+    Note: There should not be a notion of user temp dir. Instead,
+    user specified temp directories for ray should overwrite /tmp/ray.
+    """
+    logger.warning(
+        "get_user_temp_dir is deprecated, and will be removed in a future release."
+        " Please use resolve_user_ray_temp_dir instead."
+    )
+
+    return resolve_user_ray_temp_dir(None, None)
 
 
 def get_ray_temp_dir():
-    return os.path.join(get_user_temp_dir(), "ray")
+    """
+    get_ray_temp_dir is deprecated. Use resolve_user_ray_temp_dir instead.
+
+    Get the ray temp directory for the current user.
+    """
+    logger.warning(
+        "get_ray_temp_dir is deprecated, and will be removed in a future release."
+        " Please use resolve_user_ray_temp_dir instead."
+    )
+
+    return resolve_user_ray_temp_dir(None, None)
 
 
 def get_ray_address_file(temp_dir: Optional[str]):
     if temp_dir is None:
-        temp_dir = get_ray_temp_dir()
+        temp_dir = resolve_user_ray_temp_dir(None, None)
```

Comment:
Given we pass in None, this means we will only get the default temp dir not the actual temp dir of the node, is it intented?